substrate/utils/wasm-builder/src/prerequisites.rs

```diff
@@ -200,7 +200,16 @@ fn check_wasm_toolchain_installed(
 	}
 
 	let version = dummy_crate.get_rustc_version();
-	if crate::build_std_required() {
+
+	// Validate (as a sanity-check) that the target is in-fact `RuntimeTarget::Wasm`.
+	let Some(target) = RuntimeTarget::new() else {
+		return Err(colorize_error_message("Cannot compile the WASM runtime: invalid target"));
+	};
+	if target != RuntimeTarget::Wasm {
+		return Err(colorize_error_message("Cannot compile the WASM runtime: wrong target"));
+	}
```

Comment:
If you insist on checking this then this should be an assert, as the `check_wasm_toolchain_installed` should only be called when targeting WASM.

Reply:
I'll change these to asserts. I recall faintly that this was to address a review comment but could not find a reference.